src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,294 @@
+					System.out.println("\n\nThe record is "+record);	
+				}
+				else {
+					if(record.equals("http")) {
+						System.out.println("It was http");
+						String newUri = newRequest.getRequestHeader().getURI().toString();
+						bingo(getRisk(), //Risk
+								Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+								getName(), //Name
+								getDescription(), //Description
+								getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+								null, //Param
+								"", //Attack
+								Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+								getSolution(), //Solution
+								"", //Evidence
+								getCweId(), //CWE ID
+								getWascId(), //WASC ID
+								newRequest); //HTTPMessage
+					}
+					break;
+				}
+			}while(true);
+		} catch(NoRouteToHostException e) {
+			System.out.println("NoRoute");
+			String newUri = newRequest.getRequestHeader().getURI().toString();
+			bingo(getRisk(), //Risk
+					Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+					getName(), //Name
+					getDescription(), //Description
+					getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+					null, //Param
+					"", //Attack
+					Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+					getSolution(), //Solution
+					"", //Evidence
+					getCweId(), //CWE ID
+					getWascId(), //WASC ID
+					newRequest); //HTTPMessage
+			return;
+		} catch (SocketException e) {
+			System.out.println("SocketException");
+			String newUri = newRequest.getRequestHeader().getURI().toString();
+			bingo(getRisk(), //Risk
+					Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+					getName(), //Name
+					getDescription(), //Description
+					getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+					null, //Param
+					"", //Attack
+					Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+					getSolution(), //Solution
+					"", //Evidence
+					getCweId(), //CWE ID
+					getWascId(), //WASC ID
+					newRequest); //HTTPMessage
+			return; //Get out
+		} catch (SocketTimeoutException e) {
+			System.out.println("SocketTimeoutException");
+			String newUri = newRequest.getRequestHeader().getURI().toString();
+			bingo(getRisk(), //Risk
+					Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+					getName(), //Name
+					getDescription(), //Description
+					getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+					null, //Param
+					"", //Attack
+					Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+					getSolution(), //Solution
+					"", //Evidence
+					getCweId(), //CWE ID
+					getWascId(), //WASC ID
+					newRequest); //HTTPMessage
+			return; //Get out
+		} catch (SSLException e) {
+			System.out.println("SSL");
+			String newUri = newRequest.getRequestHeader().getURI().toString();
+			CharSequence s = "plaintext";
+			if(e.toString().contains(s)) {
```

Comment:
Prefer the use of `getMessage()` instead of converting to String, which might give a different (localised) message.